### ISSUE TITLE
NVDA announce wrong text in dropdown after selection

### ISSUE THREAD

**Issue by dmitriy-barkalov** [telerik/kendo-ui-core#6531]
### Bug report

NVDA announce wrong text in dropdown after selection. Same issue is for mouse selection and keyboard selection

### Reproduction of the problem

1. Open Dojo https://dojo.telerik.com/aQoRUFUM
2. Execute NVDA
3. See selected "Oranges" item in dropdown
3. Expand dropdown list
4. Click "Bananas" item

### Current behavior

NVDA reads:
Oranges 2 of 3
Oranges 1 of 1

### Expected/desired behavior

NVDA reads:
Oranges 2 of 3
Bananas 1 of 1

As we selected "Bananas" not "Oranges".

### Environment

Kendo UI version: "2020.3.1118"
Browser: Edge 92.0.902.73

**Comment by Dimitar-Goshev** [comment]
The item is tracked through aria-activedescendant attribute, whose value points to the id of a span element holding the selected item text, when the DropDownList is collapsed, or to the id of a li element, when the DropDownList is expanded. The following screenshot shows this:
![image](https://user-images.githubusercontent.com/6614183/130047224-381f32a3-99c5-4402-b743-b57277e712b6.png)

This mechanism is the same in all browsers and it works as expected.

The reported issue is related to the combination of browser and reader used. NVDA works best with Firefox, Narrator works best with the original Microsoft Edge browser (not the Chromium based one) and the Chromevox extension works best with Google Chrome and Edge Chromium. When these combinations are used for testing, the read selection is correct. But when there is a mix, e.g. Edge Chromium or Google Chrome with NVDA, the reader does not read the correct item and keeps telling the initially selected item, even though the aria-activedescendant attribute no longer points to that item and it is not written in the span. To reproduce it, follow these steps:

1. Load https://runner.telerik.io/fullscreen/aQoRUFUM in Google Chrome and start NVDA.
2. Inspect the span element holding "Oranges", then expand the dropdown and inspect the dropdown element holding "Oranges" - as a result the visible DOM elements should be as on the screenshot above.
3. Click on the DropDownList to expand it.
4. Press Down arrow - Bananas becomes selected - in the span Oranges is replaced by Bananas and the id attribute in the ul element is moved to the li of Bananas
5. Press Enter - NVDA reads Oranges, but the rendered attributes and text suggest that there is no way the DOM to be telling NVDA Oranges is still selected.

**Comment by dmitriy-barkalov** [comment]
I noticed that “valueTemplate” option been set makes DropDown behave correct with NVDA. So If I set “valueTemplate” to something, NVDA starts to read correct value from DropDown. If I not set, NVDA reads wrong value.

My assumption is that it is important to have some order between set focus, set attribute value, etc. May be “valueTemplate” config change or make more strict that order and it leads to the issue fix. I don't know Kendo UI internals but I believe same behavior could be enforced without “valueTemplate” option set.

Given the above I believe that the root cause of the issue could be on DropDown side (not in Edge or NVDA). Could you please give a second look to the issue.

**Comment by Dimitar-Goshev** [comment]
@dmitriy-barkalov Thank you for this comment. We will review it and get back to you.

**Comment by Dimitar-Goshev** [comment]
@dmitriy-barkalov We reviewed the scenario once again with valueTemplate applied - [Dojo snippet](https://dojo.telerik.com/@Dimitar-Goshev/ezAsAcuQ). What the valueTemplate does is to apply custom rendering to the span element with role="option", which holds the value of the DropDownList, when collapsed. If the valueTemplate just applies the text of the item, e.g. `valueTemplate: '#: name #'` , the rendering of the span is not changed and the issue is still reproducible. And if the valueTemplate nests the text of the item in an inner element, e.g. another span - `valueTemplate: '<span>#: name #</span>' `, then the rendering is changed.

To sum up, the issue is that NVDA with Chrome does not expected the text of an element with role="option" to be updated and speaks the old value of the element. We still consider this a reader - browser issue, as it is reproducible only with certain combinations between them.

If we remove the role="option" attribute from the span element or if we nest the text in an inner element, then NVDA speaks the updated value of the element. Removing role="option" from the span would break the normal functionality. And adding a span to the default rendering would change the rendering and could cause issues to customized DropDownLists.

We would suggest using a workaround for this scenario by setting `valueTemplate: '<span>#: name #</span>'`.
